docs/contributing/database-migrations/mssql.md

```diff
@@ -11,97 +11,72 @@ For instructions on how to apply database migrations, please refer to the
-and more. You may also modify the `.sql` files directly with any text editor.
+We recommend reading [T-SQL Code Style](../code-style/sql.md) since it has a major impact in how we
+write migrations.
 
-To make a database change, start by modifying the `.sql` files in `src/Sql/dbo`. These changes will
-also need to be applied in a migration script. Migration scripts are located in
-`util/Migrator/DbScripts`.
+:::
 
-You can either generate the migration scripts automatically using the _Schema Comparison_
-functionality or by manually writing them. Do note that the automatic method will only take you so
-far and it will need to be manually edited to adhere to the code styles.
+## SQL database project
 
-For added safe guards we have automated linting and validation to ensure the SQL project is always
-up to date with the migrations.
+The separate database definitions in `src/Sql/.../dbo` serve as a "master" reference for the
+intended and final state of the database at that time. This is crucial because the state of database
+definitions at the current moment may differ from when a migration was added in the past. These
+definitions act as a lint and validation step to ensure that migrations work as expected, and the
+separation helps maintain clarity and accuracy in database schema management and synchronization
+processes.
 
-## Modifying the database
+Additionally, a
+[SQL database project](https://learn.microsoft.com/en-us/azure-data-studio/extensions/sql-database-project-extension-sdk-style-projects)
+is in place; however, instead of using the auto-generated migrations from
+[DAC](https://learn.microsoft.com/en-us/sql/relational-databases/data-tier-applications/data-tier-applications?view=sql-server-ver16),
+we manually write migrations. This approach is chosen to enhance performance and prevent accidental
+data loss, which is why we have both a `sqlproj` and standalone migrations.
 
-Since we follow [Evolutionary Database Design _(EDD)_](./edd.mdx), any migration that modifies
-existing columns most likely needs to be split into at least two parts: a backwards compatible
-transition phase, and a non-backwards compatible phase.
+## Modifying the database
 
-### Best practices
+In accordance with the tenets of [Evolutionary Database Design](./edd.mdx) every change must be
+considered as split into two parts:
 
-When writing a migration script there are a couple of best practices we follow. Please check the
-[T-SQL Code Style][code-style-sql] for more details. But the most important aspect is ensuring the
-script can be re-run on the database multiple times without producing any errors or data loss.
+1. A backwards-compatible transition migration
+2. A non-backwards-compatible final migration
 
-### Backwards compatible
+It is likely that a change does not require a non-backwards-compatible end phase e.g. all changes
+may be backwards-compatible in their final form; in that case, only one phase of changes is
```

Comment:
One issue with our existing DB model is that we've gravitated towards making every field optional. I don't see that as a. best practice and one of the dangers of almost always only using backwards compatible migrations.

Reply:
Neither do I see it as one.